File: app/java/app/src/main/java/com/example/accounting/application/port/in/AccountUseCase.java
package com.example.accounting.application.port.in;

import com.example.accounting.domain.model.Account;
import java.util.List;

/**
 * 勘定科目ユースケース（Input Port）
 * ビジネスユースケースのインターフェース定義
 */
public interface AccountUseCase {

    /**
     * すべての勘定科目を取得
     * @return 勘定科目一覧
     */
    List<Account> getAllAccounts();

    /**
     * 科目コードで勘定科目を取得
     * @param accountCode 科目コード
     * @return 勘定科目
     */
    Account getAccountByCode(String accountCode);

    /**
     * BSPL区分で勘定科目を取得
     * @param bsplType BSPL区分（'B' または 'P'）
     * @return 勘定科目リスト
     */
    List<Account> getAccountsByBsplType(String bsplType);

    /**
     * 勘定科目を作成
     * @param account 勘定科目
     * @return 作成された勘定科目
     */
    Account createAccount(Account account);

    /**
     * 勘定科目を更新
     * @param accountCode 科目コード
     * @param account 勘定科目
     * @return 更新された勘定科目
     */
    Account updateAccount(String accountCode, Account account);

    /**
     * 勘定科目を削除
     * @param accountCode 科目コード
     */
    void deleteAccount(String accountCode);
}

File: app/scala/src/main/scala/com/example/accounting/application/port/in/AccountUseCase.scala
package com.example.accounting.application.port.in

import com.example.accounting.application.AppError
import com.example.accounting.domain.account.{Account, AccountType}

/**
 * 勘定科目ユースケース（Input Port）
 */
trait AccountUseCase:

  /** すべての勘定科目を取得 */
  def getAllAccounts(): Either[AppError, List[Account]]

  /** 勘定科目コードで取得 */
  def getAccountByCode(code: String): Either[AppError, Account]

  /** 勘定科目を作成 */
  def createAccount(account: Account): Either[AppError, Account]

  /** 勘定科目を更新 */
  def updateAccount(code: String, account: Account): Either[AppError, Account]

  /** 勘定科目を削除 */
  def deleteAccount(code: String): Either[AppError, Unit]

  /** 勘定科目種別で検索 */
  def getAccountsByType(accountType: AccountType): Either[AppError, List[Account]]
